### PR TITLE
Remove initialization of swarm but display useful error

### PR DESCRIPTION
**Pull request by NicolasMahe**
Closes https://github.com/mesg-foundation/core/issues/213

This PR remove the initialization of swarm from the container package, and display a useful error instead.

Example from a digital ocean droplet:
```
root@test-swarm-2:~# mesg-core
Docker Swarm is not initialized. Execute "docker swarm init" and try again
```
```
root@test-swarm-2:~# docker swarm init
Error response from daemon: could not choose an IP address to advertise since this system has multiple addresses on interface eth0 (128.199.243.244 and 10.15.0.8) - specify one with --advertise-addr
```
```
root@test-swarm-2:~# docker swarm init --advertise-addr 128.199.243.244
Swarm initialized: current node (sng5snnh5cm46yibdsz7bevq8) is now a manager.

To add a worker to this swarm, run the following command:

    docker swarm join --token SWMTKN-1-0kzu7i28acwa3pl0c5nsll7wo0z2xgb0pql54wlez70uwnybea-6d3kowvsni0oqztuxc7z3v2w9 128.199.243.244:2377

To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
```
```
root@test-swarm-2:~# mesg-core
MESG Core dev build test-swarm

Usage:
  mesg-core [command]

Available Commands:
  help        Help about any command
  logs        Show the Core's logs
  service     Manage services
  start       Start the Core
  status      Get the Core's status
  stop        Stop the Core

Flags:
  -h, --help         help for mesg-core
      --no-color     disable colorized output
      --no-spinner   disable spinners
      --version      version for mesg-core

Use "mesg-core [command] --help" for more information about a command.
root@test-swarm-2:~#
```

### CODE REVIEW

File: container/container.go [hunk 1]
```diff
@@ -92,22 +93,17 @@ func (c *DockerContainer) negotiateAPIVersion() {
...
 	ctx, cancel := context.WithTimeout(context.Background(), c.callTimeout)
 	defer cancel()
 	info, err := c.client.Info(ctx)
 	if err != nil {
 		return err
 	}
-	if info.Swarm.NodeID != "" {
-		return nil
+	if info.Swarm.NodeID == "" {
+		return errors.New("docker swarm is not initialized. Execute \"docker swarm init\" and try again")
```
Comment (ilgooz) [inline container/container.go]:
```suggestion
		return errors.New("docker swarm is not initialized. run \"docker swarm init\" to setup swarm and try again")
```

File: container/container.go [hunk 2]
```diff
@@ -66,7 +71,7 @@ func New(options ...Option) (*DockerContainer, error) {
 		}
 	}
 	c.negotiateAPIVersion()
-	if err := c.createSwarmIfNeeded(); err != nil {
+	if err := c.checkSwarm(); err != nil {
```
Comment (krhubert) [inline container/container.go]: Rename s/checkSwarm/ - it checks for what? 

`IsSwarmInit (bool, error)` or as You didt `IsSwarmInit error`

File: container/container_test.go [hunk 3]
```diff
@@ -93,19 +91,12 @@ func TestNew(t *testing.T) {
...
-	require.NotNil(t, c)
+	dt.ProvideInfo(types.Info{Swarm: swarm.Info{NodeID: ""}}, nil)
 
-	select {
-	case <-dt.LastSwarmInit():
-		t.Fail()
-	default:
-	}
+	_, err := New(ClientOption(dt.Client()))
+	require.Error(t, err, errSwarmNotInit)
```
Comment (krhubert) [inline container/container_test.go]: It tests if error exist and the prints msg and args.

Use Equal

File: container/container.go [hunk 4]
```diff
@@ -12,6 +13,10 @@ import (
 	"github.com/mesg-foundation/core/config"
 )
 
+var (
+	errSwarmNotInit = errors.New(`docker swarm is not initialized. run "docker swarm init" to setup swarm and try again`)
```
Comment (krhubert) [inline container/container.go]: run `"docker swarm init"` and try again.

Docker swarm init **initialize** swarm. Setup means something diffrent in this case (https://docs.docker.com/engine/swarm/swarm-tutorial/#use-docker-for-mac-or-docker-for-windows)

Also this message is short and clear

File: container/container.go (hunk 2 above)
Comment (NicolasMahe) [reply container/container.go]: what about `checkSwarmInit() error` ?

File: container/container.go (hunk 4 above)
Comment (NicolasMahe) [reply container/container.go]: Done

File: container/container.go (hunk 2 above)
Comment (krhubert) [reply container/container.go]: what's wrong with `is`? checkSwarmInit checks but for what?

you can check if is running, is init, is up etc
CheckUp, CheckRunning etc, but check alone doesn't carry any more info.

Comment (NicolasMahe) [reply container/container.go]: Rename by `isSwarmInit() error`

**Pull request merged**